backstory/src/main/java/com/google/sps/story/PromptManagerBodyFactory.java

```diff
@@ -0,0 +1,228 @@
+        if (outputBody.contains("<gerund>")) {
+          String gerund = gerunds.remove(0);
+
+          outputBody = outputBody.replaceFirst("<gerund>", gerund);
+
+          // Replace double-adjective noun case with first available noun and fetched adjectives.
+        } else if (outputBody.contains("<adj> <adj> <noun>")) {
+          String doubleAdjectiveNoun = nouns.remove(0);
+          String[] relatedAdjectives =
+              wordTools.fetchRelatedAdjectives(doubleAdjectiveNoun, 2, chooseRandomly);
+
+          doubleAdjectiveNoun =
+              relatedAdjectives[0] + " " + relatedAdjectives[1] + " " + doubleAdjectiveNoun;
+          outputBody = outputBody.replaceFirst("<adj> <adj> <noun>", doubleAdjectiveNoun);
+
+          // Replace single-adjective noun case with first available noun and fetched adjective.
+        } else if (outputBody.contains("<adj> <noun>")) {
+          String singleAdjectiveNoun = nouns.remove(0);
+          String[] relatedAdjectives =
+              wordTools.fetchRelatedAdjectives(singleAdjectiveNoun, 1, chooseRandomly);
+
+          singleAdjectiveNoun = relatedAdjectives[0] + " " + singleAdjectiveNoun;
+          outputBody = outputBody.replaceFirst("<adj> <noun>", singleAdjectiveNoun);
+
+          // Leave once tags are all replaced
+        } else {
+          break;
+        }
+      }
+
+      return outputBody;
+    } catch (Exception e) {
+      // In case of network exception, use list method.
+      return makeListTemplate();
+    }
+  }
+
+  /**
+   * Generates listing prompt body using given keywords.
+   *
+   * @return Completed prompt body.
+   */
+  private String makeListTemplate() {
+    // Use specific template if empty list is given.
+    if (keywords.size() == 0) {
+      outputBody = EMPTY_ENDING;
+      // List single keyword if given.
+    } else if (keywords.size() == 1) {
+      outputBody = "a " + keywords.get(0) + " was present.";
+      // List two keywords if given.
+    } else if (keywords.size() == 2) {
+      outputBody += "a " + keywords.get(0) + " as well as a " + keywords.get(1) + " ";
+      outputBody += getTemplate(SIMPLE_ENDINGS);
+
+    } else {
+      // Iterate through given keywords if over size 2.
+      for (int index = 0; index < keywords.size() - 1; index++) {
+        outputBody += keywords.get(index) + ", ";
+      }
+      outputBody += "as well as a " + keywords.get(keywords.size() - 1) + " ";
+
+      // Get ending template.
+      outputBody += getTemplate(SIMPLE_ENDINGS);
+    }
+    return outputBody;
+  }
+
+  /**
+   * Obtains either the first available or a randomly selected template from the list.
+   *
+   * @param templateList List of templates to choose from.
+   * @return Template obtained.
+   */
+  private String getTemplate(String[] templateList) {
+    if (chooseRandomly) {
+      return templateList[templateChooser.nextInt(templateList.length)];
+    } else {
+      return templateList[0];
+    }
+  }
+}
```

Comment:
Tests?

Reply:
I had tests for this in my PromptManagerTest.java ?